TC-DA-1.2: Fix for numberic PIDs with fallback

cecille opened the pull request:
Apparently isupper is false if there are no letters at all. Adds a fix for TC-DA-1.2 and test vector that exposes this condition.

Also adds the pid as a variable to the test vectors.

Test: test_TC_DA_1_2.py

Fixes: https://github.com/project-chip/certification-tool/issues/268

cecille committed:
TC-DA-1.2: Fix for numberic PIDs will fallback

Apparently isupper is false if there are no letters at all.
Adds a fix for TC-DA-1.2 and test vector that exposes this condition.

Also adds the pid as a variable to the test vectors.

cecille committed:
Fix test to use pid from json

cecille committed:
Merge branch 'master' into tc_da_isupper

restyled-commits committed:
Restyled by clang-format

cecille committed:
Merge branch 'master' into tc_da_isupper

tcarmelveilleux commented on src/python_testing/TC_DA_1_2.py:
```diff
@@ -56,7 +56,7 @@ def parse_single_vidpid_from_common_name(commonName: str, tag_str: str) -> str:
 
     s = sp[1][:4]
-    if not s.isupper() or len(s) != 4:
+    if not (s.isnumeric() or s.isupper()) or len(s) != 4:
```
Overall, the entire if could be replaced with `if re.match("[0-9A-F]{4}", s) is None:`

cecille replied:
Changed, added new test vector and confirmed in test that this now fails on line 81 at this function call, rather than on line 90 in the int conversion.

cecille committed:
Change to regular expression

- added a new test vector with the non-numeric PID
- confirmed in the log that the test failure now happens
  on line 81 rather than line 90. This is a manual only verification,
  I am not adjusting the test-test to expect this, any failure
  is equally valid as far as the test-test is concerned.
- remaning tests are still correct per test_TC_DA_1_2.py
